src/app/views/query-runner/query-input/QueryInput.tsx

```diff
@@ -74,43 +78,71 @@ const QueryInput = (props: IQueryInputProps) => {
-          selectedKey={sampleQuery.selectedVersion || 'v1.0'}
-          options={urlVersions}
-          onChange={(event, method) => handleOnVersionChange(method)}
-        />
+    <>
+      <div className='row' >
+        <div className='col-xs-12 col-lg-2'>
+          <Dropdown
+            ariaLabel={translateMessage('HTTP request method option')}
+            selectedKey={sampleQuery.selectedVerb}
+            options={httpMethods}
+            styles={verbSelector}
+            errorMessage={showError ? translateMessage('Sign in to use this method') : undefined}
+            onChange={(event, method) => handleOnMethodChange(method)}
+          />
+        </div>
+        <div className='col-xs-12 col-lg-2'>
+          <Dropdown
+            ariaLabel={translateMessage('Microsoft Graph API Version option')}
+            selectedKey={sampleQuery.selectedVersion || 'v1.0'}
+            options={urlVersions}
+            onChange={(event, method) => handleOnVersionChange(method)}
+          />
+        </div>
+        <div className='col-xs-12 col-lg-5'>
+          <AutoComplete
+            contentChanged={contentChanged}
+            runQuery={runQuery}
+          />
+        </div>
+        {!mobileScreen &&
+        <>
+          <div className='col-lg-2'>
+            <SubmitButton
+              className='run-query-button'
+              text={translateMessage('Run Query')}
+              disabled={showError || !sampleQuery.sampleUrl}
+              role='button'
+              handleOnClick={() => runQuery()}
+              submitting={submitting}
+              allowDisabledFocus={true}
+            />
+          </div>
+          <div className='col-lg-1' style={{flexShrink: 2}}>
+            <ShareQuery/>
+          </div>
+        </>
+        }
       </div>
-      <div className='col-xs-12 col-lg-6'>
-        <AutoComplete
-          contentChanged={contentChanged}
-          runQuery={runQuery}
-        />
+      {mobileScreen &&
+      <div style={{display: 'flex'}}>
+        <div style={{flexGrow: 5, flexBasis: '100%'}}>
+          <SubmitButton
+            className='run-query-button'
+            text={translateMessage('Run Query')}
+            disabled={showError || !sampleQuery.sampleUrl}
+            role='button'
+            handleOnClick={() => runQuery()}
+            submitting={submitting}
+            allowDisabledFocus={true}
+          />
+        </div>
+        <div style={{flexGrow: 1, flexShrink: 2}}>
+          <ShareQuery/>
+        </div>
       </div>
-      <div className='col-xs-12 col-lg-2'>
-        <SubmitButton
-          className='run-query-button'
-          text={translateMessage('Run Query')}
-          disabled={showError || !sampleQuery.sampleUrl}
-          role='button'
-          handleOnClick={() => runQuery()}
-          submitting={submitting}
-          allowDisabledFocus={true}
-        />
-      </div>
-    </div>)
+      }
+    </>
+  )
```

Comment:
This code repeats itself. It should be wrapped in a function that returns the appropriate layout depending on the screensize